chapter.md

```diff
@@ -0,0 +1,966 @@
+
+## The Payoff: Validation
+
+One way to see how these changes pays off is to look at validation,
+and in particular at the way in which we can use *selectors* to compute data by composing functions,
+and then use *memoization* to make state changes more efficient.
+
+Selectors chain functions together
+and pipe their return values into the last function in the chain.
+In the simple code below,
+the selector will return `form.character`:
+
+```ts
+const formStateSelector = (state: IAppState) => state.form;
+
+const characterFormSelector = createSelector(
+  formStateSelector,
+  (form: IForm) => form.character
+);
+```
+
+With that in hand,
+let's write a validation function that checks that all required fields are present
+and add it to our component:
+
+```ts
+export const isFormValid = createSelector(
+  characterFormSelector,
+  character => character.name
+    && character.bioSummary.age
+    && character.skills.length > 0
+);
+
+@select(isFormValidSelector)
+isFormValid$: Observable<boolean>;
+```
+
+Now,
+after selecting,
+we can use `isFormValid$` with an asynchronous pipe in our template:
+
+```ts
+<button 
+  [disabled]="!(isFormValid$ | async)"
+  type="submit">
+  Save
+</button>
+```
+
+(The `$` on the end of `isFormValid$` is a convention in the RxJS library meaning,
+"This is an observable."
+We don't have to use it,
+but we find it helps make code easier to understand.)
+
+We can now go ahead and create specific rules for specific fields of our form
+using a helper function called `isBetweenNumber` that does exactly what you'd expect:
+
+```ts
+const humanAgeValid = isBetweenNumber(14, 40);
+const elfAgeValid = isBetweenNumber(80, 800);
+const tieflingAgeValid = isBetweenNumber(35, 53);
+
+const bioSummarySelector = createSelector(
+  characterFormSelector,
+  ({bioSummary}: ICharacter) => bioSummary
+);
+
+const ageValidationSelector = createSelector(
+  bioSummarySelector,
+  (bioSummary: IBioSummary) => {
+    switch (bioSummary.race) {
+      case 'Human': return humanAgeValid;
+      case 'Elf': return elfAgeValid;
+      case 'Tiefling': return tieflingAgeValid;
+    }
+  }
+);
+
+export const isAgeValidSelector = createSelector(
+  bioSummarySelector,
+  ageValidationSelector,
+  ({age}, isAgeValid) => isAgeValid(age)
+);
+```
+
+Once we have a pile of validation rules for our fields,
+we can chain those selectors together like this:
+
+```ts
+const isFormValidSelector = createSelector(
+  isAgeValidSelector,
+  isNameValidSelector,
+  (ageValid, nameValid) => ageValid && nameValid
+);
+```
+
+In larger applications,
+we can and should go further
+and make selectors out of reusable validation functions:
+
+```ts
+const maxStringLengthValidation = (value: string, max: number) =>
+  value.length < max;
+
+const minStringLengthValidation = (value: string, min: number) =>
+  value.length > min;
+
+const isNameValidSelector = createSelector(
+  createFormFieldSelector(['character', 'name']),
```

Comment:
Please have a look at the updated state of the doc and PR if you'd like to add this in - I worry that we're already deep in the brambles.